src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -386,61 +437,72 @@ RECT NonClientIslandWindow::GetMaxWindowRectInPixels(const RECT* const prcSugges
         }
         break;
     }
+
     case WM_EXITSIZEMOVE:
     {
         ForceResize();
         break;
     }
-    case WM_NCACTIVATE:
-    case WM_NCPAINT:

```

Comment:
>case WM_NCPAINT: [](start = 4, length = 16)

What happens if we keep `WM_NCPAINT` here along with `WM_PAINT`? (completely new territory for me)

Reply:
If we handle `WM_NCPAINT` at all (by returning anything for a NCPAINT message), then the system assumes that we wanted to paint our nonclient area. Because it assumes we're doing _all_ the work, it does _nothing_. Unfortunately, the kernel is weird, so if the OS is doing nothing, the kernel still treats our window like a Windows xp/vista-like window, and the _kernel_ thinks that we have rounded top corners. The OS also won't paint our window shadow anymore, because it assumes we're doing that.

Basically everything is terrible if we try doing that.